src/dymaptic.GeoBlazor.Core/Components/Widgets/BasemapLayerListWidget.cs

```diff
@@ -0,0 +1,85 @@
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-BasemapLayerList.html">ArcGIS JS API</a>
+/// </summary>
+public class BasemapLayerListWidget : LayerListWidget
+{    
+    /// <inheritdoc />
+    [JsonPropertyName("type")]
+    public override string WidgetType => "basemapLayerList";
+
+    /// <summary>
+    ///     A delegate to implement a custom handler for setting up each<see cref="ListItem"/>.
+    ///     Function must take in a ListItem and return a Task with the same (updated) item.
+    /// </summary>
+    [Parameter]
+    [JsonIgnore]
+    public Func<ListItem, Task<ListItem>>? OnBaseListItemCreatedHandler { get; set; }
+
+    /// <summary>
+    ///     The .Net object reference to this class for calling from JavaScript.
+    /// </summary>
+    public DotNetObjectReference<BasemapLayerListWidget> BaseLayerListWidgetObjectReference => DotNetObjectReference.Create(this);
+
+    /// <summary>
+    ///     A convenience property that signifies whether a custom <see cref="OnBaseListItemCreatedHandler"/> was registered.
+    /// </summary>
+    public bool HasCustomBaseListHandler => OnBaseListItemCreatedHandler is not null;
+
+    /// <summary>
+    ///     A JavaScript invokable method that is triggered whenever a ListItem is created and a handler is attached.
+    /// </summary>
+    /// <param name="item">
+    ///     The <see cref="ListItem"/> from the original source.
+    /// </param>
+    /// <returns>
+    ///     Returns the modified <see cref="ListItem"/>
+    /// </returns>
+    [JSInvokable]
+    public Task<ListItem>? OnBaseListItemCreated(ListItem item)
+    {
+        return OnListItemCreatedHandler?.Invoke(item);
+    }
+
+    /// <summary>
+    ///     A delegate to implement a custom handler for setting up each <see cref="ListItem"/>.
+    ///     Function must take in a ListItem and return a Task with the same (updated) item.
+    /// </summary>
+    [Parameter]
+    [JsonIgnore]
+    public Func<ListItem, Task<ListItem>>? OnReferenceListItemCreatedHandler { get; set; }
+
+    /// <summary>
+    ///     The .Net object reference to this class for calling from JavaScript.
+    /// </summary>
+    //public DotNetObjectReference<BasemapLayerListWidget> ReferenceLayerListWidgetObjectReference => DotNetObjectReference.Create(this);
+
+    /// <summary>
+    ///     A convenience property that signifies whether a custom <see cref="OnReferenceListItemCreatedHandler"/> was registered.
+    /// </summary>
+    public bool HasCustomReferenceListHandler => OnReferenceListItemCreatedHandler is not null;
+
+    /// <summary>
+    ///     A JavaScript invokable method that is triggered whenever a ListItem is created and a handler is attached.
+    /// </summary>
+    /// <param name="item">
+    ///     The <see cref="ListItem"/> from the original source.
```

Comment:
Update description to clarify

Reply:
updated to reflect the reference types or base types